code/controllers/subsystem/events.dm

```diff
@@ -85,7 +85,8 @@ SUBSYSTEM_DEF(events)
 			event_roster[event_to_check] = event_to_check.weight
 
 	var/datum/round_event_control/event_to_run = pick_weight(event_roster)
-	TriggerEvent(event_to_run)
+	if(!isnull(event_to_run))
```

Comment:
this doesn't really seem to fix anything, the problem is with pick weight picking a null right?

Is that a problem or is it intentional that it can be null sometimes? In which case it needs a comment indicating this is not an unusual circumstance.

Silencing null runtimes without validating if they're actually a proper runtime or an expected side effect of things isn't proper fixing.

Reply:
The above loop selects events- it may happen that none qualify, leading to a zero-length event_roster. Imo this is a valid case. I admittedly only reproduced this on downstream Skyrat, but in testing with just tgstation code on localhost the only valid event was space dust. Since that's not valid on icebox, so I *believe* this would runtime on icebox on localhost, though I haven't tested this.

Regardless, in my opinion no events being eligible to run is an edge case that should be handled- perhaps this should log explicitly or even notify admins?

Reply:
Something like
```
if(length(event_roster) > 0)
      //pick an event and run it```
Would work  too and be more explicit- I should probably do that instead.